### PR TITLE
Enhancement: added import of .nex5 files

### PR DESCRIPTION
**Pull request by NeuroExplorer**
Added import of  .nex5 files. .nex5 is an extension of .nex data format for neurophysiological data that has been in use since 2015.

### CODE REVIEW

File: fileio/private/ft_senstype.m [hunk 1]
```diff
@@ -247,6 +247,10 @@
 elseif issubfield(input, 'orig.FileHeader') &&  issubfield(input, 'orig.VarHeader')
   % this is a complete header that was read from a Plexon *.nex file using read_plexon_nex
   type = 'plexon';
+
+elseif issubfield(input, 'orig.FileHeader') &&  issubfield(input, 'orig.VarHeaders')
```
Comment (schoffelen) [inline fileio/private/ft_senstype.m:251]: I notice that the difference between the 'plexon' and 'nex5' senstype is just one single character in one of the fieldnames of the original header structure. This is as intended?

File: fileio/private/read_nex5.m [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+%   header      structure with header information
+%   feedback    0 or 1
+%   tsonly      0 or 1, read only the timestamps and not the waveforms
+%   channel     number, or list of numbers (that will result in multiple outputs)
+%   begsample   number (for continuous only)
+%   endsample   number (for continuous only)
+%
+% See also READ_NEX5_HEADER
+%
+% Copyright (C) 2020 Robert Oostenveld
```
Comment (schoffelen) [inline fileio/private/read_nex5.m:26]: Did Robert write this? If not, feel free to add your own name here. Likewise for the other newly contributed code. Or, if the new code is based on some older code that Robert wrote, you can add both your names

**Comment by schoffelen** [comment]
Would it be possible to also provide some documentation on the website regarding the nex5 format, and motivate why to what extent this differs from the 'older' nex?

http://www.fieldtriptoolbox.org/getting_started/plexon/

Also, it would be good to provide us with a small test dataset.

File: fileio/private/ft_senstype.m (hunk 1 above)
Comment (NeuroExplorer) [reply fileio/private/ft_senstype.m:251]: I did not notice ft_senstype.m code at first and simply like VarHeaders as field name more than VarHeader. And then it was handy in ft_senstype. I can check 
orig.FileHeader.NexFileHeader == 'NEX5'
instead, for example.

File: fileio/private/read_nex5.m (hunk 2 above)
Comment (NeuroExplorer) [reply fileio/private/read_nex5.m:26]: I wrote the code myself but is is based on Robert's read_nex_* code. I will use both names.

File: fileio/private/ft_senstype.m (hunk 1 above)
Comment (schoffelen) [reply fileio/private/ft_senstype.m:251]: Ah, I see. Well I don't know who came up with the original VarHeader (so without the 's') but if this is based on a convention that is also used in other software tools, I wonder whether it's clever to adopt a slightly different fieldname for something that probably has the same content. Therefore, I would recommend sticking to the 'without-s' naming convention (even if you don't like it :) ) and do a more principled check for the 'NEX5' string.

**Comment by NeuroExplorer** [comment]
Sure, I will add information on nex5 file format.  I will also provide a couple of test nex5 data files. 

By the way, I noticed that existing read_nex_* code does not handle some .nex files properly. I will provide you with the test .nex file and fix the code in the next branch.

File: fileio/private/read_nex5.m (hunk 2 above)
Comment (schoffelen) [reply fileio/private/read_nex5.m]: Top! Nice to meet you online, Alex! Good to know who's the named person behind NeuroExplorer

**Comment by NeuroExplorer** [comment]
Hi Jan-Mathijs,

I made changes in my branch and provided links to .nex and .nex5 test files
in Github. Should I wait for Robert to review my code?

Best Regards,

Alex Kirillov <alex@neuroexplorer.com>
Nex Technologies
www.neuroexplorer.com

**Comment by schoffelen** [comment]
Thanks Alex, I reviewed your code and will merge this soon. I had to wait for our file server to be upgraded, because I want to deal with the testdata upon merging. This will happen soon! Thanks for your contributions so far, and we look forward to a fix for the old-fashioned *.nex files :)

**Comment by schoffelen** [comment]
Just to raise awareness: could the reported issue with the provided *.nex file also be related to things that are in this thread #34?

**Comment by schoffelen** [comment]
I am not inhibited by any knowledge here, because I didn't realize that the file extensions seem to be different, although the common denominator seems to be 'plexon'.

**Comment by schoffelen** [comment]
OK, merging.... Sorry for the delay

**Comment by schoffelen** [comment]
OK, it should now be in the master branch, if the overnight test batch runs through fine (this is conditional on all the accumulated new stuff in the master branch), it will percolate to the release branch as well.

**Comment by NeuroExplorer** [comment]
I found that I did not add reading .nex5 to ft_read_spike.m, so my
implementation of .nex5 import is incomplete. I will work on
ft_read_spike.m in the next few days.

**Comment by jens-k** [comment]
> Just to raise awareness: could the reported issue with the provided *.nex file also be related to things that are in this thread #34?

I saw this mentioning of the pl2/v2 thread. It's not really related right?

**Pull request merged**